packages/preset-wind/src/variants/has.ts

```diff
@@ -0,0 +1,74 @@
+import type { Variant } from '@unocss/core'
+import { h, variantGetBracket } from '@unocss/preset-mini/utils'
+
+export const variantHasPseudo: Variant[] = [
+  {
+    name: 'has',
+    match(input, ctx) {
+      const body = variantGetBracket('has-', input, ctx.generator.config.separators)
+      if (!body)
+        return
+
+      const [match, reset] = body
+      const bracketValue = h.bracket(match)
+      if (!bracketValue)
+        return
+
+      if (match) {
+        return {
+          matcher: reset,
+          handle: (input, next) => next({
+            ...input,
+            selector: `${input.selector || ''}:has(${bracketValue})`,
+          }),
+        }
+      }
+    },
+  },
+  {
+    name: 'group-has',
+    match(input, ctx) {
+      const body = variantGetBracket('group-has-', input, ctx.generator.config.separators)
+      if (!body)
+        return
+
+      const [match, reset] = body
+      const bracketValue = h.bracket(match)
+      if (!bracketValue)
+        return
+
+      if (match) {
+        return {
+          matcher: reset,
+          handle: (input, next) => next({
+            ...input,
+            selector: `.group:has(${bracketValue}) ${input.selector || ''}`,
```

Comment:
We have an option to toggle `.group` style: https://github.com/unocss/unocss/blob/5c9fc9942fcb564acc063ac4d18926a2ae14a344/packages/preset-mini/src/_variants/pseudo.ts#L254

Reply:
I have done this and moved to the `preset-mini` package, had to add some code because the previous implementation might work not perfectly with custom arguments (and I'm kinda confused with it).

And implement both four pseudo-class functions with it (so change the pr title